### ISSUE TITLE
BankHolidayOperation / DaysOfOperation only takes first value

### ISSUE THREAD

**Issue by chadselph** [planarnetwork/transxchange2gtfs#41]
Consider the following input:

```xml
      <OperatingProfile>
        <RegularDayType>
          <DaysOfWeek>
            <Sunday />
          </DaysOfWeek>
        </RegularDayType>
        <BankHolidayOperation>
          <DaysOfOperation>
            <GoodFriday />
            <LateSummerBankHolidayNotScotland />
            <MayDay />
            <EasterMonday />
            <SpringBank />
            <OtherPublicHoliday>
              <Description>Platinum Jubilee bank holiday</Description>
              <Date>2022-06-03</Date>
            </OtherPublicHoliday>
          </DaysOfOperation>
          <DaysOfNonOperation>
            <Christmas />
            <NewYearsDay />
            <ChristmasDayHoliday />
            <BoxingDayHoliday />
            <NewYearsDayHoliday />
          </DaysOfNonOperation>
        </BankHolidayOperation>
      </OperatingProfile>
```

produces this OperatingProfile object:

```javascript
{
  BankHolidayOperation: {
    DaysOfOperation: [ 'GoodFriday' ],
    DaysOfNonOperation: [ 'Christmas' ]
  },
  SpecialDaysOperation: { DaysOfOperation: [], DaysOfNonOperation: [] },
  RegularDayType: [ [
      0, 0, 0, 0,
      0, 0, 1
    ] ]
}
```

The bug is on this line which only maps over the first element. https://github.com/planarnetwork/transxchange2gtfs/blob/master/src/transxchange/TransXChangeStream.ts#L184

**Comment by linusnorton** [comment]
thanks @chadselph I'll take a look in the next few days

**Comment by linusnorton** [comment]
@chadselph this should be fixed in 1.8.5, please give it a go.
